.github/workflows/main.yml

```diff
@@ -1,35 +1,131 @@
-      uses: niden/actions-memcached@v7
-
-    - run: npm install
-    - run: npm test
-      env:
-        CI: true
+      - name: Git checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Start Redis
+        uses: supercharge/redis-github-action@1.1.0
+        with:
+          redis-version: ${{ matrix.redis-version }}
+
+      - name: Memcached Service
+        uses: niden/actions-memcached@v7
+
+      - name: Cache node_modules
+        id: cache-modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ matrix.node-version }}-${{ runner.OS }}-build-${{ hashFiles('package.json') }}
+
+      - name: Run npm install in version ${{ matrix.node-version }}
+        if: steps.cache-modules.outputs.cache-hit != 'true'
+        run: npm install
+
+      - name: Run npm test in version ${{ matrix.node-version }}
+        run: npm run test:coverage
+        env:
+          CI: true
+
+      - name: upload coverage artifact v${{ matrix.node-version }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: coveragev-${{ matrix.node-version }}
+          path: coverage/lcov.info
+
+  coverage:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+        with:
+          name: coveragev-12.x
+          path: coverage
+
+      - name: list data from directory
+        run: ls -R
+
+      - name: Generate coverage in coverall
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  build:
+    name: Build application
+    needs:
+      - coverage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+
+      - name: Cache node_modules
+        id: cache-modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: 12.x-${{ runner.OS }}-build-${{ hashFiles('package.json') }}
+
+      - name: Run npm install
+        if: steps.cache-modules.outputs.cache-hit != 'true'
+        run: npm install
+
+      - name: Run npm build
+        run: npm run build
+
+      - name: Generate dist artifact v${{ matrix.node-version }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist-folder
+          path: dist/**/*
+          if-no-files-found: error
+
+  publish:
+    name: NPM publish package
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+        with:
+          name: dist-folder
+          path: dist
+      # - uses: actions/setup-node@v1
```

Comment:
If this comments isn't necessary so remove them

Reply:
this content is the only content commited to npm package see `package.json`file rules!
this step create de asset to publish step